src/js_mixed.hpp

```diff
@@ -0,0 +1,187 @@
+    }
+};
+
+template <typename Context, typename Value, typename Utils>
+class HandleBoolean : public MixedWrapper<Context, Value> {
+    Mixed wrap(Context context, Value const &value) {
+        return Mixed(Utils::to_boolean(context, value));
+    }
+
+    Value unwrap(Context context, Mixed mixed) {
+        return Utils::from_boolean(context, mixed.get<bool>());
+    }
+};
+
+template <typename Context, typename Value, typename Utils,
+          typename RealmNumberType>
+class HandleNumber : public MixedWrapper<Context, Value> {
+    Mixed wrap(Context context, Value const &value) {
+        return Mixed(Utils::to_number(context, value));
+    }
+
+    Value unwrap(Context context, Mixed mixed) {
+        return Utils::from_number(context, mixed.get<RealmNumberType>());
+    }
+};
+
+template <typename Context, typename Value, typename Utils>
+class HandleDecimal128 : public MixedWrapper<Context, Value> {
+    Mixed wrap(Context context, Value const &value) {
+        return Mixed(Utils::to_decimal128(context, value));
+    }
+
+    Value unwrap(Context context, Mixed mixed) {
+        return Utils::from_decimal128(context, mixed.get<Decimal>());
+    }
+};
+
+template <typename Context, typename Value, typename Utils>
+class HandleObjectID : public MixedWrapper<Context, Value> {
+    Mixed wrap(Context context, Value const &value) {
+        return Mixed(Utils::to_object_id(context, value));
+    }
+
+    Value unwrap(Context context, Mixed mixed) {
+        return Utils::from_object_id(context, mixed.get<ObjectId>());
+    }
+};
+
+template <typename Context, typename Value, typename Utils>
+class HandleBinary : public MixedWrapper<Context, Value> {
+    Mixed wrap(Context context, Value const &value) {
+        auto owned_binary_data = Utils::to_binary(context, value);
+        return Mixed(owned_binary_data.get());
+    }
+
+    Value unwrap(Context context, Mixed mixed) {
+        return Utils::from_binary(context, mixed.get<BinaryData>());
+    }
+};
+
+template <typename Context, typename Value, typename Utils>
+class HandleTimeStamp : public MixedWrapper<Context, Value> {
+    Mixed wrap(Context context, Value const &value) {
+        auto date = Utils::to_date(context, value);
+
+        double milliseconds = Utils::to_number(context, date);
+        int64_t seconds = milliseconds / 1000;
+        int32_t nanoseconds = ((int64_t)milliseconds % 1000) * 1000000;
+        Timestamp ts(seconds, nanoseconds);
+
+        return Mixed(ts);
+    }
+
+    Value unwrap(Context context, Mixed mixed) {
+        return Utils::from_timestamp(context, mixed.get<Timestamp>());
+    }
+};
+
+template <typename JavascriptEngine>
+class TypeMixed {
+   private:
+    using Context = typename JavascriptEngine::Context;
+    using Value = typename JavascriptEngine::Value;
+    using Utils = js::Value<JavascriptEngine>;
+    using Strategy = MixedWrapper<Context, Value> *;
+
+    std::map<DataType, Strategy> strategies = {
```

Comment:
This seems to be missing a strategy for object or is that handled elsewhere?

Reply:
Yes that's correct but that's something we plan to support in the next release.